### PR TITLE
feature: allowed shutdowning a direction when there is a light thread…

### PR DESCRIPTION
**Pull request by spacewander**
… blocking on it.

I hereby granted the copyright of the changes in this pull request
to the authors of this lua-resty-core project.

### CODE REVIEW

**Comment by spacewander** [comment]
This is the sister PR of https://github.com/openresty/lua-nginx-module/pull/1516

File: lib/ngx/pipe.md [hunk 1]
```diff
@@ -314,9 +314,8 @@ a string describing the error.
 If the `merge_stderr` option is specified in [spawn](#spawn), closing the
 `stderr` direction will return `nil` and the error string `"merged to stdout"`.
 
-Shutting down a direction when a light thread is waiting on it (such as during
-reading or writing) will return `nil` and the error string `"pipe busy
-writing"` (for stdin) or `"pipe busy reading"` (for the others).
+Shutting down a direction when there is a light thread waiting on it (like
+reading or writing) will abort the light thread and return `true`.
```
Comment (agentzh) [inline lib/ngx/pipe.md]: This English sentence has grammatical errors.

Comment (spacewander) [reply lib/ngx/pipe.md]: @agentzh 
Fixed in the new commit.

**Comment by thibaultcha** [comment]
Merged, thanks!

**Pull request closed without merging**